flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1264,6 +1273,483 @@ def test_all_transitions(self, desired_state, discovered_state):
+                    dataset_id=ScenarioMixin.DATASET_ID,
+                    metadata=ScenarioMixin.METADATA,
+                ),
+            ],
+        )
+
+    def test_deleted_non_manifest_dataset(self):
+        """
+        If there is a manfestation on this node that is deleted,
+        the corresponding dataset has a desired state of ``DELETED``.
+        """
+        assert_desired_datasets(
+            self, self.deployer,
+            desired_manifestations=[
+                ScenarioMixin.MANIFESTATION.transform(
+                    ["dataset", "deleted"], True,
+                    ['dataset', 'metadata'], ScenarioMixin.METADATA,
+                ),
+            ],
+            local_datasets=[
+                DiscoveredDataset(
+                    dataset_id=ScenarioMixin.DATASET_ID,
+                    blockdevice_id=ScenarioMixin.BLOCKDEVICE_ID,
+                    state=DatasetStates.NON_MANIFEST,
+                    maximum_size=int(REALISTIC_BLOCKDEVICE_SIZE.bytes),
+                ),
+            ],
+            expected_datasets=[
+                DesiredDataset(
+                    state=DatasetStates.DELETED,
+                    dataset_id=ScenarioMixin.DATASET_ID,
+                    metadata=ScenarioMixin.METADATA,
+                ),
+            ],
+        )
+
+    def test_leased_mounted_manifestaion(self):
+        """
+        If there is a lease for a mounted dataset present on node,
+        there is a corresponding dataset that has a desired state of
+        ``MOUNTED``.
+        """
+        assert_desired_datasets(
+            self, self.deployer,
+            desired_manifestations=[],
+            local_datasets=[
+                DiscoveredDataset(
+                    dataset_id=ScenarioMixin.DATASET_ID,
+                    blockdevice_id=ScenarioMixin.BLOCKDEVICE_ID,
+                    state=DatasetStates.MOUNTED,
+                    maximum_size=int(REALISTIC_BLOCKDEVICE_SIZE.bytes),
+                    device_path=FilePath('/dev/xvdf'),
+                    mount_point=FilePath('/mount/path'),
+                )
+            ],
+            expected_datasets=[
+                DesiredDataset(
+                    state=DatasetStates.MOUNTED,
+                    dataset_id=ScenarioMixin.DATASET_ID,
+                    maximum_size=int(REALISTIC_BLOCKDEVICE_SIZE.bytes),
+                    metadata={},
+                    mount_point=self.deployer.mountroot.child(
+                        unicode(ScenarioMixin.DATASET_ID)
+                    ),
+                ),
+            ],
+            leases=Leases().acquire(
+                now=datetime.now(tz=UTC),
+                dataset_id=ScenarioMixin.DATASET_ID,
+                node_id=self.deployer.node_uuid,
+            )
+        )
+
+    def test_leased_attached_manifestaion(self):
+        """
+        If there is a lease for an attached dataset present on node,
+        there is not a corresponding desired dataset.
```

Comment:
Add explanation that this is in context of not mentioning dataset in configuration.